src/expression.js

```diff
@@ -705,17 +705,24 @@ pp.parseFunctionBody = function(node, isArrowFunction) {
     // Start a new scope with regard to labels and the `inFunction`
     // flag (restore them to their old value afterwards).
     let oldLabels = this.labels
+    let oldVarDeclaredNames = this.varDeclaredNames
+    this.varDeclaredNames = new Set()
```

Comment:
We're targeting ES3, so though we use ES6 syntax, we can't use ES5/6 runtime objects. This'll have to be done with a plain object and `Object.prototype.hasOwnProperty`.